lib/introspect.js

```diff
@@ -20,24 +22,34 @@ module.exports.processXML = function(err, xml, obj, callback) {
     if (err) return callback(err);
     if (!result.node) throw new Error('No root XML node');
     result = result.node; // unwrap the root node
-    // If no interface, try first sub node?
-    if (!result.interface) {
-      if (result.node && result.node.length > 0 && result.node[0]['$']) {
-        var subObj = Object.assign(obj, {});
-        if (subObj.name.slice(-1) !== '/') subObj.name += '/';
-        subObj.name += result.node[0]['$'].name;
-        return module.exports.introspectBus(subObj, callback);
-      }
-      return callback(new Error('No such interface found'));
-    }
     var proxy = {};
     var nodes = [];
     var ifaceName, method, property, iface, arg, signature, currentIface;
-    var ifaces = result['interface'];
+    // If there are no interfaces available then at least insert the
+    // Introspectable interface so that users can do their own introspection
+    // via `getObject`.
+    var ifaces = result['interface'] || [
```

Comment:
I'm a bit skeptical of this approach because it advertises that the interface is there when it may actually not be, and it will not be added when there are other interfaces present. If it's true that interfaces might be present and not advertised, I think a better solution would be to be able to call any method and just get an error if it's not there. I think that's how the other libraries work, but it would take some research to confirm that.